File: training-data/objectivec/PowerOfTwo.m
#import <Foundation/Foundation.h>

BOOL isPowerOfTwo(NSInteger n) {
    return n > 0 && (n & (n - 1)) == 0;
}

int main(int argc, const char *argv[]) {
    @autoreleasepool {
        NSLog(@"%d", isPowerOfTwo(16));
        NSLog(@"%d", isPowerOfTwo(18));
    }
    return 0;
}

File: training-data/prolog/power_of_two.pl
is_power_of_two(N) :-
    N > 0,
    X is N /\ (N - 1),
    X =:= 0.

:- (is_power_of_two(16) -> writeln(true) ; writeln(false)).
:- (is_power_of_two(18) -> writeln(true) ; writeln(false)).
